source/common/conn_pool/conn_pool_base.cc

```diff
@@ -600,6 +609,28 @@ void ActiveClient::onConnectTimeout() {
   close();
 }
 
+void ActiveClient::onLifetimeTimeout() {
```

Comment:
onConnectionDurationTimeout would be more consistent.  ditto on the timer and the logs.

Reply:
I originally wanted to be consistent with the max streams variable name (which refers to  "lifetime") but it probably makes more sense to be consistent with the API field names.